File: ruby/huifier.rb
def huify(word)
  word = word.downcase.strip
  vowels = 'аеёиоуыэюя'
  rules = {
    'а' => 'я',
    'о' => 'ё',
    'у' => 'ю',
    'ы' => 'и',
    'э' => 'е',
  }
  word.split('').each do |letter|
    if vowels.include?(letter) then
      if rules.key?(letter) then
        word = rules[letter] + word[1..-1]
      end
      break
    else
      word = word[1..-1]
    end
  end
  return word.empty? ? 'Хуй' : 'Ху' + word
end

File: python/huifier.py
def huify(word):
    word = word.lower().strip()
    vowels = 'аеёиоуыэюя'
    rules = {
        'а': 'я',
        'о': 'ё',
        'у': 'ю',
        'ы': 'и',
        'э': 'е',
    }
    for letter in word:
        if letter in vowels:
            if letter in rules:
                word = rules[letter] + word[1:]
            break
        else:
            word = word[1:]
    return 'Ху' + word if word else 'Хуй'

if __name__ == '__main__':
    print('Введите слово:')
    print(huify(input()))
